### PR TITLE
[UPX i12] Enable UPX i12 basic boot

### PR DESCRIPTION
**Pull request by tsaikevin**
Add support for Up Xtreme i12 ADLP based board.
The PCIe M.2 slot CN12 on the board is able to detect NVMe SSD. Debug output is enabled at header CN9 on the board(e.g. UART1) Tested to boot with OS loader payload and UEFI payload.

To stitch the SlimBootloader.bin with IFWI uses StitchLoader.py script with '-p' as given below:

python Platform/AlderlakeBoardPkg/Script/StitchLoader.py -i <BIOS_IMAGE_NAME> -s Outputs/adlp/SlimBootloader.bin -o sbl_upx12_ifwi.bin -p 0xAA000104

### CODE REVIEW

**Comment by sindhuragrandhi** [comment]
Please make sure to update the year in the header. May I know why 0xAA000104 is needed to be patched?

**Comment by gdong1** [comment]
> Please make sure to update the year in the header. May I know why 0xAA000104 is needed to be patched?
0xAA000104  is the stitch data defined as below. It set the PlatformID to 4 we could not read BoardId from EC, SMBUS or GPIO.
typedef struct {
  UINT8               PlatformId : 5;
  UINT8               Reserved1  : 3;
  UINT8               DebugUart;
  UINT8               Reserved3;
  UINT8               Marker;
} STITCH_DATA;

**Review by gdong1 — APPROVED** [review]

File: Silicon/AlderlakePkg/Library/IgdOpRegionLib/IgdOpRegionLib.c [hunk 1]
```diff
@@ -293,6 +293,7 @@ UpdateVbt (
   case PLATFORM_ID_ADL_P_LP4_RVP:
   case PLATFORM_ID_ADL_P_LP5_RVP:
   case PLATFORM_ID_RPL_P_DDR5_CRB:
+  case PLATFORM_ID_ADL_P_UPXI12:
     DEBUG((DEBUG_INFO, "UpdateVbt: BoardIdAdlP DDR5 or Lp4/5Rvp .....\n"));
```
Comment (kvparikh89) [inline Silicon/AlderlakePkg/Library/IgdOpRegionLib/IgdOpRegionLib.c:297]: Update debug print

**Comment by tsaikevin** [comment]
> > Please make sure to update the year in the header. May I know why 0xAA000104 is needed to be patched?
> > 0xAA000104  is the stitch data defined as below. It set the PlatformID to 4 we could not read BoardId from EC, SMBUS or GPIO.
> > typedef struct {
> > UINT8               PlatformId : 5;
> > UINT8               Reserved1  : 3;
> > UINT8               DebugUart;
> > UINT8               Reserved3;
> > UINT8               Marker;
> > } STITCH_DATA;

@Guo,

this patch is needed for UP board that SBL cannot read board ID,
in this case, I assign Platform ID 4, and set debug port to 1 for this board.

you may reference the same setup for i11 TGL one, https://slimbootloader.github.io/supported-hardware/upxtremei11.html

**Comment by tsaikevin** [comment]
> Please make sure to update the year in the header. May I know why 0xAA000104 is needed to be patched?

@sindhuragrandhi, fixed the header, pls check again. thanks

**Review by kvparikh89 — APPROVED** [review]
Looks good to me

**Comment by sindhuragrandhi** [comment]
LGTM

**Pull request merged**